client/src/pages/Courses/CourseView.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from "react";
+import { Flex, Text } from "@chakra-ui/core";
+import { FieldsPane, RequirementsPane } from "./editables";
+
+const fields = {
+    code: "courseCode",
+    name: "name",
+    desc: "description",
+    sem: "semester",
+    pts: "points",
+};
+
+const CourseView = ({ course, isNew, isEditing, onEdit, onDelete, cancelUpdateCourse, updateCourse }) => {
+    const [code, setCode] = useState("");
+    const [name, setName] = useState("");
+    const [desc, setDesc] = useState("");
+    const [sem, setSem] = useState([]);
+    const [pts, setPts] = useState(15);
+
+    const setField = {
+        code: setCode,
+        name: setName,
+        desc: setDesc,
+        sem: setSem,
+        pts: setPts
+    };
+
+    useEffect(() => {
+        setCode(course.courseCode || "");
+        setName(course.name || "");
+        setDesc(course.description || "");
+        setSem(course.semester || []);
+        setPts(course.points || 15);
+    }, [course])
+
+    const changeField = (field, value) => {
+        setField[field](value);
+    }
+
+    const saveCourse = () => {
+        const editedCourse = { ...course };
+        editedCourse[fields.code] = code;
```

Comment:
Could destructure here but lgtm

Reply:
wdym?

Reply:
like {...course, courseCode: code, name: name, .......} ?

Reply:
yea. just realised that fields.code is not the same as code so de-structuring wont work. my bad